packages/driver/src/cy/commands/querying/within.ts

```diff
@@ -76,6 +78,10 @@ export default (Commands, Cypress, cy, state) => {
       })
     }
 
+    // TODO: Rework cy.within to use chainer-based subject chaining, rather than its custom withinSubject state.
+    // For now, we leave this logic in place and just ensure that the new rules don't interfere with it.
+    cy.breakSubjectLinksToCurrentChainer()
```

Comment:
Could you explain this? If I am understanding correctly, it seem like this would break the following? 


```
cy.get('.form').within(() => { do something}).should('contain', 'submit')
```

Reply:
So the key is that cy.within works differently from cy.then - it *never changes the subject.* According to our docs (https://docs.cypress.io/api/commands/within#Yields):

>  .within() yields the same subject it was given from the previous command.
> Trying to return a different element the .within callback have no effect:
```
cy.get('#within-yields')
  .within(() => {
    // we are trying to return something
    // from the .within callback,
    // but it won't have any effect
    return cy.contains('Child element').should('have.class', 'some-child')
  })
  .should('have.id', 'within-yields')
```

So cy.within always matches the two conditions I mentioned in my other comment - the callback contains cypress commands, and we do not want to use their resolved subject as the new subject of the parent chainer. The existing automated tests back up this behavior.

I didn't know this about cy.within - I learned it by looking at the tests, being confused, then looking at the docs to see if the tests were correct. :rofl: I can't speak to why it works this way, just that it does and I'm trying to retain all the existing rules.

Reply:
Yeah, `.within()` is a bit interesting in that regard -- but curious what these new rules are exactly? It seems this logic is maintaining parity with the current logic?

Does your TODO comment mean you'd like to change this behavior later on or that you need to remove `cy.breakSubjectLinksToCurrentChainer()` ?

Reply:
Yes, this logic is maintaining parity with existing functionality. The 'two rules' I mentioned are just guidelines to help understand when `breakSubjectLinksToCurrentChainer` is appropriate to use.

The TODO comment is that I'll need to refactor `cy.within` as part of the effort of implementing selectors / fixing detached DOM. The goal will still be to maintain the existing functionality as part of any further work in here.